src/components/contexts/error-handler.tsx

```diff
@@ -0,0 +1,85 @@
+import { DialogContent, Divider, Typography } from "@mui/material";
+import { useAtom } from "jotai";
+import { ReactNode, useMemo } from "react";
+import { errorAtom } from "../../atoms/errorAtom";
+import GenericDialog from "../generics/generic-dialog";
+
+/**
+ * Component properties
+ */
+interface Props {
+  children: ReactNode;
+}
+
+/**
+ * Error handler component
+ *
+ * @param props component properties
+ */
+const ErrorHandler = ({ children }: Props) => {
+  const [error, setError] = useAtom(errorAtom);
+
+  /**
+   * Handles error message
+   *
+   * @param message error message
+   * @param err any error
+   */
+  const handleError = async (message: string, err?: any) => {
```

Comment:
Can we type err with something more specific than any?

Reply:
Typed it as an object